### PR TITLE
Update rust devcontainer.json

### PR DESCRIPTION
**Pull request by midnightexigent**
- use [rust-analyzer](https://marketplace.visualstudio.com/items?itemName=matklad.rust-analyzer) for better linting and auto-complete 
- add [crates](https://marketplace.visualstudio.com/items?itemName=serayuzgur.crates) extension for better experience when editing `Cargo.toml`
- use [Even Better TOML](https://marketplace.visualstudio.com/items?itemName=tamasfe.even-better-toml) for an even better experience when editing `Cargo.toml`

### CODE REVIEW

File: containers/rust/.devcontainer/devcontainer.json [hunk 1]
```diff
@@ -1,34 +1,37 @@
...
-	"settings": { 
+	"settings": {
 		"terminal.integrated.shell.linux": "/bin/bash",
 		"lldb.executable": "/usr/bin/lldb",
 		// VS Code don't watch files under ./target
 		"files.watcherExclude": {
 			"**/target/**": true
-		}
+		},
+		"rust-analyzer.checkOnSave.command": "clippy"
```
Comment (Aisuko) [inline containers/rust/.devcontainer/devcontainer.json:21]: Should we add the notation of `rust-analyzer` for giving a choice to the user? Just like `forwardPorts` below, and I searched for `clippy`. And I found many people have other views https://internals.rust-lang.org/t/should-rust-analyzer-run-clippy-by-default/12766, Would you mind explaining more about this? I'd like to know more about this on my personal.

File: containers/rust/.devcontainer/devcontainer.json [hunk 2]
```diff
@@ -1,34 +1,37 @@
...
 	},
-
 	// Add the IDs of extensions you want installed when the container is created.
 	"extensions": [
-		"rust-lang.rust",
-		"bungcip.better-toml",
 		"vadimcn.vscode-lldb",
-		"mutantdino.resourcemonitor"
+		"mutantdino.resourcemonitor",
```
Comment (Aisuko) [inline containers/rust/.devcontainer/devcontainer.json:26]: I have s suggestion here, we can sort the extensions by what's features they provide and follow the sort of  `rust-server` -> `optimist toml` -> `debug tools` -> `others` would be good.

**Review by Aisuko — CHANGES_REQUESTED** [review]
Thanks for contributing and there may have some advice, please kindly take a look.

File: containers/rust/.devcontainer/devcontainer.json (hunk 1 above)
Comment (midnightexigent) [reply containers/rust/.devcontainer/devcontainer.json:21]: Fine by me to add the notation of `rust-analyzer` for giving a choice to the user. 

As for `clippy` vs`check`, `cargo clippy` itself runs `cargo check` but also gives additional warnings. Using as much warnings as possible generally results in more stable and better code. Though, I would agree that making it the default is opinionated.

File: containers/rust/.devcontainer/devcontainer.json (hunk 2 above)
Comment (midnightexigent) [reply containers/rust/.devcontainer/devcontainer.json:26]: Fine by me

Comment (Aisuko) [reply containers/rust/.devcontainer/devcontainer.json:26]: Thank you.

File: containers/rust/.devcontainer/devcontainer.json (hunk 1 above)
Comment (Aisuko) [reply containers/rust/.devcontainer/devcontainer.json:21]: Thank you

**Comment by 2percentsilk** [comment]
Hey there!
Even Better TOML is a relatively recent extension that doesn't have a high level of usage yet. How necessary do you think it is? 
If we could leave it out for now, we're happy to add the other two extensions to the definition.

**Comment by midnightexigent** [comment]
Hello 

> How necessary do you think it is?

At the time I made the PR, Even better TOML was the only TOML extension that had autocomplete for Cargo.toml files. Meaning that it was the only extension that had schema support. Other extensions didn't seem to do much other that syntax highlight (I could be wrong about this)

**Comment by zebp** [comment]
Giving this a bump, switching from RLS to rust-analyzer would make using the official devcontainer a smoother experience.

**Review by 2percentsilk — APPROVED** [review]

**Comment by Aisuko** [comment]
hi @midnightexigent ,  I guess there may need a `git rebase`, let's do this?  I believe this is the final step that the PR get merge.

**Comment by midnightexigent** [comment]
@Aisuko 

Done

**Comment by Aisuko** [comment]
Hi, @midnightexigent Thanks a lot.

**Comment by Aisuko** [comment]
Hi @2percentsilk, Would you please help to merge this?

File: containers/rust/.devcontainer/devcontainer.json [hunk 3]
```diff
@@ -1,33 +1,37 @@
...
-
+	"runArgs": [
+		"--cap-add=SYS_PTRACE",
+		"--security-opt",
+		"seccomp=unconfined"
+	],
 	// Set *default* container specific settings.json values on container create.
-	"settings": { 
+	"settings": {
+		"terminal.integrated.shell.linux": "/bin/bash",
```
Comment (NewGyu) [inline containers/rust/.devcontainer/devcontainer.json:15]: `terminal.integrated.shell.*` has been deprecated.
https://code.visualstudio.com/updates/v1_56#_terminal

I think this line should be rewritten as the below:
```json
		"terminal.integrated.profiles.linux": {
			"bash": {
				"path": "/bin/bash",
			}
		},
		"terminal.integrated.defaultProfile.linux": "bash",
```

Comment (eitsupi) [reply containers/rust/.devcontainer/devcontainer.json:15]: FYI, in other container definitions, the option was simply removed. #860

**Comment by midnightexigent** [comment]
I marked the changes as resolved, is that sufficient ?

**Review by 2percentsilk — APPROVED** [review]

**Pull request merged**